rules/core/suggestions.js

```diff
@@ -1,150 +1,289 @@
+    allow: [
+      'arrowFunctions',
+      'functions',
+      'methods',
+    ],
+  }],
   'no-empty-static-block': 'off',
   'no-eq-null': 'off',
-  'no-eval': 'off',
-  'no-extend-native': 'off',
-  'no-extra-bind': 'off',
-  'no-extra-boolean-cast': 'off',
-  'no-extra-label': 'off',
-  'no-extra-semi': 'off',
-  'no-floating-decimal': 'off',
-  'no-global-assign': 'off',
+  'no-eval': 'error',
+  'no-extend-native': 'error',
+  'no-extra-bind': 'error',
+  'no-extra-boolean-cast': 'error',
+  'no-extra-label': 'error',
+  'no-extra-semi': 'error',
+  'no-floating-decimal': 'error',
+  'no-global-assign': ['error', { exceptions: [] }],
   'no-implicit-coercion': 'off',
   'no-implicit-globals': 'off',
-  'no-implied-eval': 'off',
+  'no-implied-eval': 'error',
   'no-inline-comments': 'off',
   'no-invalid-this': 'off',
-  'no-iterator': 'off',
-  'no-label-var': 'off',
-  'no-labels': 'off',
-  'no-lone-blocks': 'off',
-  'no-lonely-if': 'off',
-  'no-loop-func': 'off',
+  'no-iterator': 'error',
+  'no-label-var': 'error',
+  'no-labels': ['error', { allowLoop: false, allowSwitch: false }],
+  'no-lone-blocks': 'error',
+  'no-lonely-if': 'error',
+  'no-loop-func': 'error',
   'no-magic-numbers': 'off',
-  'no-mixed-operators': 'off',
-  'no-multi-assign': 'off',
-  'no-multi-str': 'off',
+      'no-mixed-operators': ['error', { groups: [    ['&', '|', '^', '~', '<<', '>>', '>>>'],    ['==', '!=', '===', '!==', '>', '>=', '<', '<='],    ['&&', '||'],    ['in', 'instanceof']  ],  allowSamePrecedence: true}],
+  'no-multi-assign': 'error',
+  'no-multi-str': 'error',
   'no-negated-condition': 'off',
-  'no-nested-ternary': 'off',
-  'no-new': 'off',
-  'no-new-func': 'off',
-  'no-new-object': 'off',
-  'no-new-wrappers': 'off',
+  'no-nested-ternary': 'error',
+  'no-new': 'error',
+  'no-new-func': 'error',
+  'no-new-object': 'error',
+  'no-new-wrappers': 'error',
   'no-nonoctal-decimal-escape': 'off',
-  'no-octal': 'off',
-  'no-octal-escape': 'off',
-  'no-param-reassign': 'off',
-  'no-plusplus': 'off',
-  'no-proto': 'off',
-  'no-redeclare': 'off',
-  'no-regex-spaces': 'off',
+  'no-octal': 'error',
+  'no-octal-escape': 'error',
+  'no-param-reassign': ['error', {
+    props: true,
+    ignorePropertyModificationsFor: [
+      'acc', // for reduce accumulators
+      'accumulator', // for reduce accumulators
+      'e', // for e.returnvalue
+      'ctx', // for Koa routing
+      'req', // for Express requests
+      'request', // for Express requests
+      'res', // for Express responses
+      'response', // for Express responses
+      '$scope', // for Angular 1 scopes
+      'staticContext', // for ReactRouter context
+    ],
+  }],
+  'no-plusplus': 'error',
+  'no-proto': 'error',
+  'no-redeclare': 'error',
+  'no-regex-spaces': 'error',
   'no-restricted-exports': 'off',
-  'no-restricted-globals': 'off',
+      'no-restricted-globals': ['error', 'isFinite', 'isNaN'].concat(confusingBrowserGlobals),
```

Comment:
Do we need to import `const confusingBrowserGlobals = require('confusing-browser-globals');`?